src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -250,19 +253,10 @@ public void index(SinkRecord record, DocWriteRequest<?> request) {
       throw error.get();
     }
 
-    if (docIdToRecord != null) {
-      // make sure that only unique records are being sent in a batch
-      // every request that is flushed and succeeds triggers a callback that removes it from the map
-      while (docIdToRecord.containsKey(request.id())) {
-        flush();
-        clock.sleep(TimeUnit.SECONDS.toMillis(1));
-      }
-    }
-
     // wait for internal buffer to be less than max.buffered.records configuration
     long maxWaitTime = clock.milliseconds() + config.flushTimeoutMs();
     while (numRecords.get() >= config.maxBufferedRecords()) {
-      clock.sleep(TimeUnit.SECONDS.toMillis(1));
+      clock.sleep(WAIT_TIME);
```

Comment:
Is there a reason why this wait time was chosen? Would this value change if there were x times more requests?
I suppose perf tests will give more backing to what the value would be 

Could be overkill but this made me think of setting a [rate limiter](https://guava.dev/releases/22.0/api/docs/index.html?com/google/common/util/concurrent/RateLimiter.html) on `addToRequestToRecordMap`. Suppose we hit the rate limit, we increase the rate limit as well as the wait time accordingly to better performance

Reply:
we actually don't want to rate limit, we want to run as fast as possible and only wait if we dont have enough buffer space

Reply:
I see 👍🏽